starknet-core/src/types/state_update.rs

```diff
@@ -27,6 +27,9 @@ pub struct StateDiff {
     pub deployed_contracts: Vec<DeployedContract>,
     #[serde_as(as = "Vec<UfeHex>")]
     pub declared_contracts: Vec<FieldElement>,
+    #[serde(default)]
+    #[serde_as(as = "HashMap<UfeHex, UfeHex>")]
+    pub nonces: HashMap<FieldElement, FieldElement>,
```

Comment:
Looks like we're missing a test case where this field is not empty. Am looking for a block with nonce changes and will come back shortly.